x-pack/plugins/lens/public/indexpattern_datasource/operations/layer_helpers.ts

```diff
@@ -240,34 +237,244 @@ export function replaceColumn({
+
+        // Try to reuse the previous field by finding a possible operation
+        if (
+          !hasFieldMatch &&
+          'sourceField' in previousColumn &&
+          validation.input.includes('field')
+        ) {
+          const defIgnoringfield = operationDefinitions.filter(
+            (def) =>
+              def.input === 'field' &&
+              isOperationAllowedAsReference({ validation, operationType: def.type, indexPattern })
+          );
+
+          // No exact match found, so let's determine that the current field can be reused
+          const defWithField = defIgnoringfield.filter((def) => {
+            if (def.input !== 'field') return;
+            const previousField = indexPattern.getFieldByName(previousColumn.sourceField);
+            if (!previousField) return;
+            return isOperationAllowedAsReference({
+              validation,
+              operationType: def.type,
+              field: previousField,
+              indexPattern,
+            });
+          });
+
+          if (defWithField.length === 1) {
+            // Found an exact field and operation match,
+            hasFieldMatch = true;
+            promotedOutputLayer = insertNewColumn({
+              layer: promotedOutputLayer,
+              columnId: newId,
+              op: defWithField[0].type,
+              indexPattern,
+              field: indexPattern.getFieldByName(previousColumn.sourceField),
+            });
+            return newId;
+          } else if (defIgnoringfield.length === 1) {
+            hasFieldMatch = true;
+            promotedOutputLayer = {
+              ...promotedOutputLayer,
+              incompleteColumns: {
+                ...promotedOutputLayer.incompleteColumns,
+                [newId]: { operationType: defIgnoringfield[0].type },
+              },
+            };
+            return newId;
+          } else if (defWithField.length > 0) {
+            hasFieldMatch = true;
+
+            const newColumns = { ...promotedOutputLayer.columns };
+            delete newColumns[columnId];
+
+            promotedOutputLayer = {
+              ...promotedOutputLayer,
+              // this logic produces the wrong order, but is fixed in the next step
+              columnOrder: [
+                ...promotedOutputLayer.columnOrder.map((id) => (id === columnId ? newId : id)),
+                columnId,
+              ],
+              incompleteColumns: {
+                ...promotedOutputLayer.incompleteColumns,
+                [newId]: { sourceField: previousColumn.sourceField },
+              },
+            };
+            promotedOutputLayer.columnOrder = getColumnOrder(promotedOutputLayer);
+            promotedOutputLayer.columns = adjustColumnReferencesForChangedColumn(
+              promotedOutputLayer,
+              newId
+            );
+            return newId;
+          }
+        }
+
+        // The reference is too ambiguous at this point, but instead of throwing an error
```

Comment:
About the transition from e.g. "Moving average of average of bytes to cumulative sum of bytes" - this would probably the place to put another condition in to check the references of the previous columns to contain fields that could be used.

I'm aware there could be multiple, but as the alternative is to error out, we could just take the first field that fits.